### ISSUE TITLE
Make donor users more generic

### ISSUE THREAD

**Issue by Kyhl** [pollopollo-org/back-end#145]
Currently, donor users support anonymous creating and login, but they do not support non-anonymous login. 
This creates an array of problems since making the front-end generic becomes very hard since a donor user currently lacks a number of fields (First name, Surname, Country) that a regular user has. 

If we were to implement these fields in the donor user as optional parameters, i think that the front-end would be way more loose and extendable, and would allow us to re-use much of the already existing code, as long as we can extend the basic user-model always.

So, in short:

* Add optional fields to a donoruser - and by extension donormodel - which allows it to be substituted for a usermodel.
* Minor refactoring of the backend to support these parameters.
* Could be a slight change to the front-end to support the input of these parameters when creating a user.
